# HG changeset patch
# Parent 326246896e5ea324a46d2d6360e6f9bcc6fc587e

diff --git a/gpu/config/software_rendering_list_json.cc b/gpu/config/software_rendering_list_json.cc
--- a/gpu/config/software_rendering_list_json.cc
+++ b/gpu/config/software_rendering_list_json.cc
@@ -240,16 +240,22 @@ const char kSoftwareRenderingListJson[] 
       ]
     },
     {
       "id": 24,
       "description": "Accelerated 2d canvas is unstable in Linux at the moment",
       "os": {
         "type": "linux"
       },
+      "exceptions": [
+        {
+          "gl_vendor": "Qualcomm",
+          "gl_renderer": "Adreno.*320"
+        }
+      ],
       "features": [
         "accelerated_2d_canvas"
       ]
     },
     {
       "id": 27,
       "description": "ATI/AMD cards with older drivers in Linux are crash-prone",
       "cr_bugs": [95934, 94973, 136240, 357314],
