Description: Add OxideMsgStart to IPCMessageStart enum
Author: Chris Coulson <chris.coulson@canonical.com>

diff --git a/ipc/ipc_message_start.h b/ipc/ipc_message_start.h
--- a/ipc/ipc_message_start.h
+++ b/ipc/ipc_message_start.h
@@ -87,12 +87,13 @@ enum IPCMessageStart {
   WebSocketMsgStart,
   NaClHostMsgStart,
   WebRTCIdentityMsgStart,
   LocalDiscoveryMsgStart,
   PowerMonitorMsgStart,
   EncryptedMediaMsgStart,
   ServiceWorkerMsgStart,
   MessagePortMsgStart,
+  OxideMsgStart,
   LastIPCMsgStart  // Must come last.
 };
 
 #endif  // IPC_IPC_MESSAGE_START_H_
