Description: Add OxideMsgStart to IPCMessageStart enum
Author: Chris Coulson <chris.coulson@canonical.com>

diff --git a/ipc/ipc_message_start.h b/ipc/ipc_message_start.h
--- a/ipc/ipc_message_start.h
+++ b/ipc/ipc_message_start.h
@@ -83,12 +83,13 @@ enum IPCMessageStart {
   WebRtcLoggingMsgStart,
   TtsMsgStart,
   MemoryBenchmarkMsgStart,
   WebSocketMsgStart,
   NaClHostMsgStart,
   WebRTCIdentityMsgStart,
   LocalDiscoveryMsgStart,
   PowerMonitorMsgStart,
+  OxideMsgStart,
   LastIPCMsgStart      // Must come last.
 };
 
 #endif  // IPC_IPC_MESSAGE_START_H_
