Index: oxide/chromium/src/ipc/ipc_message_start.h
===================================================================
--- oxide.orig/chromium/src/ipc/ipc_message_start.h	2013-08-16 00:01:30.143131939 +0100
+++ oxide/chromium/src/ipc/ipc_message_start.h	2013-08-16 00:01:30.143131939 +0100
@@ -88,6 +88,7 @@
   WebRTCIdentityMsgStart,
   LocalDiscoveryMsgStart,
   PowerMonitorMsgStart,
+  OxideMsgStart,
   LastIPCMsgStart      // Must come last.
 };
 
