diff --git a/tools/gritsettings/resource_ids b/tools/gritsettings/resource_ids
--- a/tools/gritsettings/resource_ids
+++ b/tools/gritsettings/resource_ids
@@ -213,9 +213,12 @@
   },
   "components/component_strings.grd": {
     "messages": [30000],
   },
   "third_party/WebKit/public/blink_resources.grd": {
     "includes": [30500],
   },
   # Resource ids starting at 31000 are reserved for projects built on Chromium.
+  "oxide_resources.grd": {
+    "includes": [31000],
+  },
 }
