Description: Make gyp work correctly when toplevel != depth
Author: Chris Coulson <chris.coulson@canonical.com>

Index: oxide/chromium/src/tools/gyp/pylib/gyp/generator/make.py
===================================================================
--- oxide.orig/chromium/src/tools/gyp/pylib/gyp/generator/make.py	2013-07-15 07:52:36.818444781 +0100
+++ oxide/chromium/src/tools/gyp/pylib/gyp/generator/make.py	2013-07-23 19:19:50.140610663 +0100
@@ -90,7 +90,11 @@
     default_variables.setdefault('OS', operating_system)
     default_variables.setdefault('SHARED_LIB_SUFFIX', '.so')
     default_variables.setdefault('SHARED_LIB_DIR','$(builddir)/lib.$(TOOLSET)')
-    default_variables.setdefault('LIB_DIR', '$(obj).$(TOOLSET)')
+    offset = '.'
+    if 'options' in params:
+      offset = os.path.relpath(params['options'].depth,
+                               params['options'].toplevel_dir)
+    default_variables.setdefault('LIB_DIR', os.path.join('$(obj).$(TOOLSET)', offset))
 
 
 def CalculateGeneratorInputInfo(params):
Index: oxide/chromium/src/tools/gyp/pylib/gyp/__init__.py
===================================================================
--- oxide.orig/chromium/src/tools/gyp/pylib/gyp/__init__.py	2013-07-12 20:54:58.790487377 +0100
+++ oxide/chromium/src/tools/gyp/pylib/gyp/__init__.py	2013-07-23 19:20:27.420610807 +0100
@@ -197,7 +197,7 @@
   set that as appropriate.
   """
   def FixPath(path):
-    path = gyp.common.FixIfRelativePath(path, options.depth)
+    path = gyp.common.FixIfRelativePath(path, options.toplevel_dir)
     if not path:
       return os.path.curdir
     return path
