# HG changeset patch
# Parent 7e09b5b0a6f5c346bafd78128742700f41b3fa4d
# User oxide

diff --git a/ui/events/events.gyp b/ui/events/events.gyp
--- a/ui/events/events.gyp
+++ b/ui/events/events.gyp
@@ -82,16 +82,17 @@
       'target_name': 'events',
       'type': '<(component)',
       'dependencies': [
         '<(DEPTH)/base/base.gyp:base',
         '<(DEPTH)/base/third_party/dynamic_annotations/dynamic_annotations.gyp:dynamic_annotations',
         '<(DEPTH)/skia/skia.gyp:skia',
         '../gfx/gfx.gyp:gfx',
         '../gfx/gfx.gyp:gfx_geometry',
+        'devices/events_devices.gyp:events_devices',
         'events_base',
         'gesture_detection',
       ],
       'defines': [
         'EVENTS_IMPLEMENTATION',
       ],
       'sources': [
         # Note: sources list duplicated in GN build.
