Description: Support library versioning in gyp
Author: Chris Coulson <chris.coulson@canonical.com>

diff --git a/tools/gyp/pylib/gyp/generator/make.py b/tools/gyp/pylib/gyp/generator/make.py
--- a/tools/gyp/pylib/gyp/generator/make.py
+++ b/tools/gyp/pylib/gyp/generator/make.py
@@ -1307,16 +1307,18 @@ class MakefileWriter:
         target = target[3:]
       target_prefix = 'lib'
       target_ext = '.a'
     elif self.type in ('loadable_module', 'shared_library'):
       if target[:3] == 'lib':
         target = target[3:]
       target_prefix = 'lib'
       target_ext = '.so'
+      if 'shared_library_version' in spec:
+        target_ext = target_ext + '.' + spec['shared_library_version']
     elif self.type == 'none':
       target = '%s.stamp' % target
     elif self.type != 'executable':
       print ("ERROR: What output file should be generated?",
              "type", self.type, "target", target)
 
     target_prefix = spec.get('product_prefix', target_prefix)
     target = spec.get('product_name', target)
diff --git a/tools/gyp/pylib/gyp/input.py b/tools/gyp/pylib/gyp/input.py
--- a/tools/gyp/pylib/gyp/input.py
+++ b/tools/gyp/pylib/gyp/input.py
@@ -73,16 +73,17 @@ base_non_configuration_keys = [
   'libraries',
   'postbuilds',
   'product_dir',
   'product_extension',
   'product_name',
   'product_prefix',
   'rules',
   'run_as',
+  'shared_library_version',
   'sources',
   'standalone_static_library',
   'suppress_wildcard',
   'target_name',
   'toolset',
   'toolsets',
   'type',
 
