Description: Add oxide::MessagePump to base::RunLoop friends
Author: Chris Coulson <chris.coulson@canonical.com>

Index: oxide/chromium/src/base/run_loop.h
===================================================================
--- oxide.orig/chromium/src/base/run_loop.h	2013-08-15 23:56:09.175131581 +0100
+++ oxide/chromium/src/base/run_loop.h	2013-08-15 23:56:09.175131581 +0100
@@ -10,6 +10,10 @@
 #include "base/memory/weak_ptr.h"
 #include "base/message_loop/message_loop.h"
 
+namespace oxide {
+class MessagePump;
+}
+
 namespace base {
 #if defined(OS_ANDROID)
 class MessagePumpForUI;
@@ -74,6 +78,7 @@
 
  private:
   friend class MessageLoop;
+  friend class oxide::MessagePump;
 #if defined(OS_ANDROID)
   // Android doesn't support the blocking MessageLoop::Run, so it calls
   // BeforeRun and AfterRun directly.
