diff -r 355f3315053b content/public/renderer/content_renderer_client.h
--- a/content/public/renderer/content_renderer_client.h	Fri Nov 28 16:07:15 2014 +1000
+++ b/content/public/renderer/content_renderer_client.h	Fri Nov 28 16:12:52 2014 +1000
@@ -49,10 +49,15 @@
 class WebWorkerPermissionClientProxy;
 struct WebPluginParams;
 struct WebURLError;
+class WebMediaPlayer;
+class WebFrame;
+class WebMediaPlayerClient;
 }
 
 namespace media {
 struct KeySystemInfo;
+class WebMediaPlayerDelegate;
+class MediaLog;
 }
 
 namespace content {
@@ -288,6 +293,13 @@
   // Returns a user agent override specific for |url|, or empty string if
   // default user agent should be used.
   virtual std::string GetUserAgentOverrideForURL(const GURL& url);
+
+  // Allow overriding default media player
+  virtual blink::WebMediaPlayer*  OverrideWebMediaPlayer(
+              blink::WebFrame* frame,
+              blink::WebMediaPlayerClient* client,
+              base::WeakPtr<media::WebMediaPlayerDelegate> delegate,
+              media::MediaLog* media_log) { return 0; }
 };
 
 }  // namespace content
diff -r 355f3315053b content/renderer/render_frame_impl.cc
--- a/content/renderer/render_frame_impl.cc	Fri Nov 28 16:07:15 2014 +1000
+++ b/content/renderer/render_frame_impl.cc	Fri Nov 28 16:12:52 2014 +1000
@@ -1728,6 +1728,17 @@
 #if defined(OS_ANDROID)
   return CreateAndroidWebMediaPlayer(url, client, initial_cdm);
 #else
+
+  blink::WebMediaPlayer* webMediaPlayer = 
+    GetContentClient()->renderer()->OverrideWebMediaPlayer(
+        frame_,
+        client,
+        weak_factory_.GetWeakPtr(),
+        new RenderMediaLog());
+
+  if (webMediaPlayer != 0)
+    return webMediaPlayer; 
+
   RenderThreadImpl* render_thread = RenderThreadImpl::current();
   media::WebMediaPlayerParams params(
       base::Bind(&ContentRendererClient::DeferMediaLoad,
