# HG changeset patch
# Parent 73e49ebd39b2a37d17d2a56ae14d68c220d01346
# User oxide

diff --git a/content/content_browser.gypi b/content/content_browser.gypi
--- a/content/content_browser.gypi
+++ b/content/content_browser.gypi
@@ -1343,24 +1343,26 @@
       'browser/web_contents/aura/window_slider.h',
       'browser/web_contents/touch_editable_impl_aura.cc',
       'browser/web_contents/touch_editable_impl_aura.h',
       'browser/web_contents/web_contents_android.cc',
       'browser/web_contents/web_contents_android.h',
       'browser/web_contents/web_contents_impl.cc',
       'browser/web_contents/web_contents_impl.h',
       'browser/web_contents/web_contents_view.h',
-      'browser/web_contents/web_contents_view_android.cc',
-      'browser/web_contents/web_contents_view_android.h',
-      'browser/web_contents/web_contents_view_aura.cc',
-      'browser/web_contents/web_contents_view_aura.h',
+      #'browser/web_contents/web_contents_view_android.cc',
+      #'browser/web_contents/web_contents_view_android.h',
+      #'browser/web_contents/web_contents_view_aura.cc',
+      #'browser/web_contents/web_contents_view_aura.h',
       'browser/web_contents/web_contents_view_guest.cc',
       'browser/web_contents/web_contents_view_guest.h',
-      'browser/web_contents/web_contents_view_mac.h',
-      'browser/web_contents/web_contents_view_mac.mm',
+      #'browser/web_contents/web_contents_view_mac.h',
+      #'browser/web_contents/web_contents_view_mac.mm',
+      '<(DEPTH)/../../../shared/port/content_browser/web_contents_view_oxide.cc',
+      '<(DEPTH)/../../../shared/port/content_browser/web_contents_view_oxide.h',
       'browser/web_contents/web_drag_dest_mac.h',
       'browser/web_contents/web_drag_dest_mac.mm',
       'browser/web_contents/web_drag_source_mac.h',
       'browser/web_contents/web_drag_source_mac.mm',
       'browser/web_contents/web_drag_utils_win.cc',
       'browser/web_contents/web_drag_utils_win.h',
       'browser/webui/content_web_ui_controller_factory.cc',
       'browser/webui/content_web_ui_controller_factory.h',
